File: Multimi/Subiect16/subiect16_micronaut_java/src/main/java/com/sd/laborator/PerechiResponse.java
package com.sd.laborator;

import io.micronaut.core.annotation.Introspected;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

@Introspected
public class PerechiResponse {
    private String message;
    private List<Integer> aValues = new ArrayList<>();
    private List<Integer> bValues = new ArrayList<>();
    private List<Map<String, Integer>> perechi = new ArrayList<>();

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public List<Integer> getAValues() {
        return aValues;
    }

    public void setAValues(List<Integer> aValues) {
        this.aValues = aValues;
    }

    public List<Integer> getBValues() {
        return bValues;
    }

    public void setBValues(List<Integer> bValues) {
        this.bValues = bValues;
    }

    public List<Map<String, Integer>> getPerechi() {
        return perechi;
    }

    public void setPerechi(List<Map<String, Integer>> perechi) {
        this.perechi = perechi;
    }
}

File: Multimi/Subiect16/perechi-gradle/src/main/kotlin/com/sd/laborator/PerechiResponse.kt
package com.sd.laborator

import io.micronaut.core.annotation.Introspected

/**
 * Corpul raspunsului HTTP.
 *   - adtA     : cele 100 de valori din ADT-ul A
 *   - adtB     : cele 100 de valori din ADT-ul B
 *   - perechi  : lista de perechi (a, b) cu a*b == a + b*3
 *   - message  : mesaj descriptiv
 */
@Introspected
class PerechiResponse {
    private var message: String? = null
    private var adtA: List<Int>? = null
    private var adtB: List<Int>? = null
    private var perechi: List<Map<String, Int>>? = null

    fun getMessage(): String? = message
    fun setMessage(m: String?) { message = m }

    fun getAdtA(): List<Int>? = adtA
    fun setAdtA(v: List<Int>?) { adtA = v }

    fun getAdtB(): List<Int>? = adtB
    fun setAdtB(v: List<Int>?) { adtB = v }

    fun getPerechi(): List<Map<String, Int>>? = perechi
    fun setPerechi(v: List<Map<String, Int>>?) { perechi = v }
}
